app/src/main/java/comp3350/mealbuddy/objects/goals/Goal.java

```diff
@@ -9,8 +9,19 @@
     public GoalType goalType;
 
     public Goal(int lowerBound, int upperBound, GoalType goalType) {
+        validateData(lowerBound, upperBound, goalType);
         this.lowerBound = lowerBound;
         this.upperBound = upperBound;
         this.goalType = goalType;
     }
+
```

Comment:
/*
 * validateData
 * verifies the data and throws the corresponding exceptions
 */